app/models/agents/weather_agent.rb

```diff
@@ -135,98 +108,82 @@ def validate_location
     end
 
     def validate_options
-      errors.add(:base, "service must be set to 'darksky' or 'wunderground'") unless wunderground? || dark_sky?
       validate_location
       errors.add(:base, "api_key is required") unless interpolated['api_key'].present?
       errors.add(:base, "which_day selection is required") unless which_day.present?
     end
 
-    def wunderground
-      if key_setup?
-        forecast = Wunderground.new(interpolated['api_key'], language: language.upcase).forecast_for(location)
-        merged = {}
-        forecast['forecast']['simpleforecast']['forecastday'].each { |daily| merged[daily['period']] = daily }
-        forecast['forecast']['txt_forecast']['forecastday'].each { |daily| (merged[daily['period']] || {}).merge!(daily) }
-        merged
-      end
-    end
-
     def dark_sky
       if key_setup?
         ForecastIO.api_key = interpolated['api_key']
         lat, lng = coordinates
-        ForecastIO.forecast(lat, lng, params: {lang: language.downcase})['daily']['data']
```

Comment:
Why did you remove the language argument?

Reply:
I removed the language argument because it was causing the request to fail.

Reply:
Can you please check it again? Their API docs still list the `lang` parameter https://darksky.net/dev/docs#forecast-request

Reply:
Yeah I can check it again.

Reply:
I added in the `language` parameter and it did work.

Reply:
That is great!